samtranslator/model/stepfunctions/events.py

```diff
@@ -167,13 +186,12 @@ def _construct_target(self, resource, role, provided_role=None, dead_letter_queu
         )
 
         if provided_role:
-            role_arn = fnSub("arn:aws:iam::${AWS::AccountId}:role/${Role}", {"Role": provided_role})
             target = {
                 "Arn": resource.get_runtime_attr("arn"),
                 "Id": target_id,
-                "RoleArn": role_arn,
+                "RoleArn": provided_role,
             }
-        else:
+        elif role:
```

Comment:
```suggestion
  target = {
      "Arn": resource.get_runtime_attr("arn"),
      "Id": target_id,
      "RoleArn": role_arn,
      "RoleArn": provided_role if provided_role else role.get_runtime_attr("arn"),
  }
```
Think this is cleaner

Reply:
I tried doing that but the linting fails then because roles type is Optional and it basically requires us to explicitly check that role is defined. A way around that is that we can generate a dummy IAMRole and pass that in, but then it makes less sense to read from the outside.